### ISSUE TITLE
Implement map_call method

### ISSUE THREAD

**Issue by vanyarock01** [tarantool/vshard#255]
Everyone who needs to execute a call on all shards writes their map_call function. Something like that:
```lua
local function map_callro(fn_name, replicaset_map)
    local callro_handler = function(replicaset, args)
        local res, err = replicaset:callro(fn_name, args)
        if err ~= nil then
            error(err)
        end
        return res
    end

    local replicasets = vshard.router.routeall()
    local fiber_map = {}
    for uuid, args in pairs(replicaset_map) do
        fiber_map[uuid] = fiber.new(callro_handler, replicasets[uuid], args)
        fiber_map[uuid]:set_joinable(true)
    end

    local res_map = { --[[ [uuid] = <result of replicaset:callro>, ]] }
    local err_map = { --[[ [uuid] = <error>, ]] }

    for uuid, fiber_obj in pairs(fiber_map) do
        local ok, data = fiber_obj:join()
        if ok then
            res_map[uuid] = data
        else
            err_map[uuid] = data
        end
    end
    return res_map, err_map
end

local res, errs = map_callro('foo', {
    ['12bb449e-6342-4e8f-bf99-ec23e9daefb1'] = {--[[ call args ]]},
    ['b4cc3aa0-2cb1-46cb-aa4f-910bb16666e3'] = {--[[ call args ]]},
})
```
Maybe add this to vshard?

**Comment by Gerold103** [comment]
Duplicate of https://github.com/tarantool/vshard/issues/147.

**Comment by hackallcode** [comment]
In the specified issue, only `map_callrw` was made, but `map_callro` was not made, which is missing when using vshard. So, I am reopening this ticket with a request to make `map_callro`. I propose to do with the creation of fibers to make `map_callro` parallel

**Comment by olegrok** [comment]
> @Gerold103, in the specified issue, only `map_callrw` was made, but `map_callro` was not made, which is missing when using vshard. So, I am reopening this ticket with a request to make `map_callro`. I propose to do with the creation of fibers to make `map_callro` parallel

You propose incorrect solution. Please read https://github.com/tarantool/vshard/discussions/261 before.

> The function has rw suffix, because can only work on masters so far. Perhaps it will be possible to improve, when https://github.com/tarantool/vshard/issues/173 is fixed. Now there is no way to prevent bucket move from a replica.

So it's impossible to implement reliable enough map_callro due to #173

**Comment by Gerold103** [comment]
I closed this ticket because it is a duplicate, not because I don't want to implement `map_callro`. I also don't see a link between RO requests being unreliable right now and creation of fibers. `map_callrw` works in parallel too, but without any new fibers.
Anyway, I created a separate ticket for RO map: https://github.com/tarantool/vshard/issues/333.
